Code refactoring

alinebiath opened the pull request:
Improvements to replace variables names, remove unnecessary comments and reorganize code.

alinebiath committed:
removendo logs, sleeps e comentarios desnecessarios

alinebiath committed:
Criacao da keyword Scroll to

alinebiath committed:
Criacao da keyword Scroll to specific position e atualizacao do seu uso no codigo.

alinebiath committed:
reorganizando codigo: criação das seções Actions,Validations e Commom

alinebiath committed:
Inserindo screenshots nas validações

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -77,23 +77,15 @@ I proceed to checkout
-    #executa o scroll de 0 até a posição - 550
-    selenium.Execute Javascript                                                      window.scrollTo(0,${total_position_y})
+    Set Test Variable                                                                ${locator}                                                        xpath=//*[@id="center_column"]/p[2]/a[1]
+    Scroll to specifc Position                                                       ${locator}
```
@alinebiath I suggest undo this change and add a comment explaining the motivation for that workaround.

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -77,23 +77,15 @@ I proceed to checkout
-    ${total_position_y}                                                              Evaluate                                                          ${total_position_y} - 550
-    #executa o scroll de 0 até a posição - 550
-    selenium.Execute Javascript                                                      window.scrollTo(0,${total_position_y})
+    Set Test Variable                                                                ${locator}                                                        xpath=//*[@id="center_column"]/p[2]/a[1]
```
@alinebiath, What library alias this keyword belongs to?

alinebiath replied:
Instated Builtin library.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -1,9 +1,9 @@
 *** Settings ***
-# para testar um aplicação web
+# para testar um aplicação web.
```
@alinebiath I suggest add a tag/table Documentation and put all url address for each library. What do you think?

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -90,18 +79,22 @@ I proceed to checkout
     selenium.Wait Until Page Contains Element                                        xpath=//*[@id="center_column"]                                    timeout=8s
+    #obter a posição horizontal do botão "Proceed to checkout"
     ${total_position_y}                                                              selenium.Get Horizontal Position                                  xpath=//*[@id="center_column"]/p[2]/a[1]
+    # subtrai 550 da posição obtida
```
This 550 represents an offset? It's a workaround/trick for some problem?

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -65,10 +65,8 @@ I add to shopping cart the listed product
-    ${btn_position_y}                                                                selenium.Get Horizontal Position                                  xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]
-    #executar comando de scroll de 0 até a posíção do botão "Add to cart"
-    selenium.Execute Javascript                                                      window.scrollTo(0,${btn_position_y})
+    Set Test Variable                                                                ${locator}                                                        xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]
```
What is the alias for that keywork?

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
 I select the sub-category "${product}"
-    selenium.Wait Until Element is Visible                                           xpath://*[@id="block_top_menu"]/ul/li[1]/ul                       timeout=8s
-    selenium.Click Element                                                           link:${product}
+    selenium.Wait Until Element is Visible                                           xpath=//*[@id="block_top_menu"]/ul/li[1]/ul                             timeout=12s
```
What do think about configure a new default timeout to entire test case?

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
-    selenium.Wait Until Element is Visible                                           xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]       timeout=8s
-    Set Test Variable                                                                ${locator}                                                        xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]
+    selenium.Wait Until Element is Visible                                           xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]             timeout=8s
+    Set Test Variable                                                                ${locator}                                                              xpath=//*[@id="center_column"]/ul/li/div/div[2]/div[2]/a[1]
```
What is the alias for that keyword?

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
-    selenium.Wait Until Page Contains Element                                        xpath=//*[@id="center_column"]                                    timeout=8s
-    Set Test Variable                                                                ${locator}                                                        xpath=//*[@id="center_column"]/p[2]/a[1]
+    selenium.Wait Until Page Contains Element                                        xpath=//*[@id="center_column"]                                          timeout=8s
+    Set Test Variable                                                                ${locator}                                                              xpath=//*[@id="center_column"]/p[2]/a[1]
```
Alias.

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
     ${last_name}=                                                                    faker.Last Name
-    selenium.Input Text                                                              id=customer_lastname                                              ${last_name}
+    selenium.Input Text                                                              id=customer_lastname                                                    ${last_name}
     ${passwd}=                                                                       faker.Password
```
Would be password instead of passwd?

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
     ${birth_day}=                                                                    faker.Day Of Month
-    ${birth_day_ns}=                                                                 string.Strip String                                               ${birth_day}                                                   mode=left           characters=0
-    selenium.Select From List By Value                                               id=days                                                           ${birth_day_ns}
+    ${birth_day_ns}=                                                                 string.Strip String                                                     ${birth_day}                                                   mode=left           characters=0
```
Please, improve that variable name.

alinebiath replied:
Done.

jaelsonwagner commented on resource/keywords_bdd.robot:
```diff
@@ -14,132 +14,134 @@ ${BROWSER}    firefox
     ${birth_month}=                                                                  faker.Month
-    ${birth_month_ns}=                                                               string.Strip String                                               ${birth_month}                                                 mode=left           characters=0
-    selenium.Select From List By Value                                               id=months                                                         ${birth_month_ns}
+    ${birth_month_ns}=                                                               string.Strip String                                                     ${birth_month}                                                 mode=left           characters=0
```
Please, improve that variable name.

alinebiath replied:
Done.

alinebiath committed:
Undo changes in keyword "Scroll to specific position"

	Deleting commom keyword and applying directly workaround to make scroll
	in a specific case.
	Case: on the cart page, normal scroll don't run correctly.
	It was necessary to obtain the horizontal position and
	subtract by 550.

alinebiath committed:
Created an alias to imported library.

	Alias "std" was created to identify Builtin library.

alinebiath committed:
Parsing variable names.

alinebiath committed:
Standardizing timeout.

	Setting 8s how standard to wait load page items.
